jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/BufferedResponseHandler.java

```diff
@@ -197,125 +208,36 @@ public void write(boolean last, ByteBuffer byteBuffer, Callback callback)
                     ConnectionMetaData connectionMetaData = getRequest().getConnectionMetaData();
                     ByteBufferPool bufferPool = connectionMetaData.getConnector().getByteBufferPool();
                     boolean useOutputDirectByteBuffers = connectionMetaData.getHttpConfiguration().isUseOutputDirectByteBuffers();
-                    _accumulator = new CountingByteBufferAccumulator(bufferPool, useOutputDirectByteBuffers, getBufferSize());
+                    _bufferedContentSink = Content.Sink.asBuffered(getWrapped(), bufferPool, useOutputDirectByteBuffers, getBufferSize());
```

Comment:
Perhaps make a public static method on Response interface that does:
```java
    public Content.Sink asBufferedSink(Request request, Response response)
    {
        ConnectionMetaData connectionMetaData = request.getConnectionMetaData();
        ByteBufferPool bufferPool = connectionMetaData.getConnector().getByteBufferPool();
        boolean useOutputDirectByteBuffers = connectionMetaData.getHttpConfiguration().isUseOutputDirectByteBuffers();
        return Content.Sink.asBuffered(response, bufferPool, useOutputDirectByteBuffers, getBufferSize());
    }
```

And then this becomes (without the lines above):
```suggestion
                    _bufferedContentSink = Response.asBufferedSink(getRequest(), getWrapped());
```

I think this static method would be useful for direct users of buffered sinks that do not want to work out all the parameters of `Content.Sink.asBuffered(...)` themselves.

Reply:
Done, but I added an extra `Response.asBufferedSink(req, rsp, int bufferSize, boolean direct)` method as `BufferedResponseHandler` can control the buffer size from a request attribute.